model_helper.py

```diff
@@ -129,6 +129,14 @@ def compute_loss_sigmoid(logits, targets, final_sequence_length, target_sequence
 
     return loss
 
+def compute_log_probs_loss(outputs):
+    nfeatures = outputs.shape[-1] // 2
+    log_prob_ones = outputs[..., :nfeatures]
+    log_prob_zeros = outputs[..., nfeatures:]
+    norm_const = tf.stop_gradient(-(log_prob_ones + log_prob_zeros) / 2)
+    loss = tf.abs((tf.exp(log_prob_ones + norm_const) + tf.exp(log_prob_zeros + norm_const)) / tf.exp(norm_const) - 1)
```

Comment:
This is equivalent to `1/p1 + 1/p0 -1`? So this loss attempts to keep probs near 0 and 1?

Reply:
it is equivalent to enforcing `p1 + p2 = 1`. Here `norm_const` is for numerical stability.

Reply:
Ah! I missed that stop gradient part! It is a bit counter-intuitive that the gradients correspond to `p1 + p0 - 1`, but the actual loss value corresponds to `1/p1 + 1/p0 - 1`. And with zero gradients you can have a loss equal to arbitrary number >= 3.

Reply:
How about adding `tf.stop_gradient(-1/p1 - 1/p0 + p1 + p0)` to loss? This way loss and gradients would correspond to each other.

Reply:
`stop_gradient` is applied to a normalization constant. It depends on `p1` and `p2` just to get proper value for each output. It is assumed to be constant, and is just used for numerical stability, and no gradients should pass through it. In other words, we might set it as `1e6`, for example, but it may not work for some outputs.

Reply:
We multiply and divide equation `p1 + p0 = 1` by `e^norm_const`, so the actual loss remains `p1 + p0 - 1`.

Reply:
You are right, my mistake.